jscomp/others/jsxPPXReactSupportC.res

```diff
@@ -26,12 +26,12 @@ module Jsx = JsxC
 
 %%private(
   @val
-  external propsWithKey: (@as(json`{}`) _, 'props, {"key": string}) => 'props = "Object.assign"
+  external propsWithKey: ({"key": string}, 'props) => 'props = "Object.assign"
```

Comment:
This was intentional to avoid mutating the object.

Reply:
Am I remembering correctly? @cknitt

Reply:
Before:
1. Create `empty obj`
2. Create `obj with key`
3. Mixing `props` and `obj with key` to the `empty object`

After:
1. Create `obj with key`
2. Mixing `props` to the `obj with key`

Reply:
If I remember correctly, we can't use key when defining a component, but even so, wouldn't it be more proper implementation to do Object.assign props with key object in order?

Reply:
Then it'll mutate the `props` object.

Reply:
I meant `Object.assign({}, props, {"key": key})`. That's why the empty object was the first argument as I remember.

Reply:
I don't understand what you want to say 😅

Reply:
Sorry for the lack of explanation.
I meant I think that the existing implementation is more appropriate.

What do you think? @cknitt

Reply:
This should work like the previous implementation (no mutation), but with one less object allocation. Looks good to me.

Reply:
(Provided that the caller passes a different {"key": key} object every time, but that seems to be the case and this is an internal function anyway.)

Reply:
Okay. It seems unlikely that `Object.assign({"key": key}, props)` would change the value of key at runtime, since props would not be able to contain key when the JSX ppx transforms the component application.

LGTM.